gamemaster/members.php

```diff
@@ -614,22 +614,139 @@ function registerNMRs($nmrs) {
 		}
 	}
 	
+	private $activeNMRs = false;
 	/**
-	 * Reduces the excuses of all members with NMRs and set members with no excuses
-	 * as left. 
+	 * Check if any active NMRs (i.e. NMRs by members with status 'playing') 
+	 * were detected during NMR handling.
+	 * 
+	 * @return boolean Returns true, if there is at least one NMR by an active member.
+	 */
+	function withActiveNMRs() {
+		return $this->activeNMRs;
+	}
+	
+	/**
+	 * Handle NMRs and check, if further sanctions due to unexcused NMRs have 
+	 * to be imposed.
 	 */
 	function handleNMRs() {
+		global $DB;
 		
+		/**
+		* Check if there is at least one active NMR and for that case reduce the 
+		* excuses of all active members with NMRs and set members with no excuses
+		* as left.
+		*/
+		$this->activeNMRs = false;
 		foreach( $this->ByStatus['Playing'] as $Member ){
-			
+
 			if( $Member->missedPhases == 0 ) continue; // no NMR
+
+			$this->activeNMRs = true; // there is at least one active NMR
 			
 			if( $Member->excusedMissedTurns > 0 ){
 				$Member->removeExcuse();
 			} else {
 				$Member->setLeft();
 			}
 		}
+		
+		/*
+		 * For all player with status left, that NMRed this turn, the NMR is always
+		 * counted as unexcused. An unexcused turn might impose temp bans as
+		 * further sanctions.
+		 */
+		foreach( $this->ByStatus['Left'] as $Member ) {
+			
+			if( $Member->missedPhases == 0 ) continue; // no NMR
+			
+			/*
+			 * Check if the NMR got an excuse and count the number of unexcused NMRs 
+			 * during the last year for the member to decide what to do.
+			 */
+			list( $systemExcused, $modExcused, $samePeriodExcused ) = 

```

Comment:
Does this need a limit 1 in case they get put into left, aren't temp banned, come back, and get temp banned again, we just want the top record right?

Reply:
sql_row() only retrieves the first row anyway. But adding a limit 1 will surely improve readability.